lib/puppet/provider/network_snmp/cisco_nexus.rb

```diff
@@ -0,0 +1,42 @@
+require 'cisco_node_utils'
+
+# Implementation for the network_snmp type using the Resource API.
+class Puppet::Provider::NetworkSnmp::CiscoNexus
+  def get(context)
+    @network_snmp ||= Cisco::SnmpServer.new
+
+    current_state = {
+      name:     'default',
+      enable:   (@network_snmp.protocol? || context.device.facts['operatingsystem'] == 'ios_xr') ? true : false,
+      contact:  @network_snmp.contact.empty? ? 'unset' : @network_snmp.contact,
+      location: @network_snmp.location.empty? ? 'unset' : @network_snmp.location,
+    }
+
+    [current_state]
+  end
+
+  def set(context, changes)
+    changes.each do |name, change|
+      should = change[:should]
+      context.updating(name) do
+        update(context, name, should)
+      end
+    end
+  end
+
+  def create(_context, name, should)
+    raise Puppet::ResourceError, "Can't create new SNMP server settings '#{name}' with #{should.inspect}"
+  end
+
+  def update(context, name, should)
+    context.notice("Updating '#{name}' with #{should.inspect}")
+    @network_snmp ||= Cisco::SnmpServer.new
+    @network_snmp.contact = should[:contact] == 'unset' ? '' : should[:contact]
+    @network_snmp.location = should[:location] == 'unset' ? '' : should[:location]
+    @network_snmp.protocol = should[:enable] unless context.device.facts['operatingsystem'] == 'ios_xr'
```

Comment:
My understanding is this module will only manage `nexus` devices, as there are no `xr` devices for `nexus`, which means this check would be redundant? 

If there is potential for this module to manage `ios_xr` devices then https://github.com/puppetlabs/cisco-network-puppet-module/blob/puppet6/lib/puppet/util/network_device/cisco_nexus/device.rb#L27 would need changes to use the `platform.image_version` rather than being hardcoded.

Reply:
I'd initially tried to ensure feature parity with agentful.

However following our questions with Rick, it seems XR is dropped, I shall remove this and the appropriate tests.